x-pack/plugins/apm/server/lib/index_pattern/create_static_index_pattern.ts

```diff
@@ -33,10 +31,8 @@ export async function createStaticIndexPattern(
 
   try {
     const apmIndexPatternTitle = config['apm_oss.indexPattern'];
-    const internalSavedObjectsClient = getInternalSavedObjectsClient(
-      context.__LEGACY.server
-    );
-    await internalSavedObjectsClient.create(
+    const savedObjectsClient = context.core.savedObjects.client;
```

Comment:
I changed this to a scoped saved object client, which used to be internal before. i tested it with the apm read user and it was able to persist the index pattern for discover/kuery bar/etc.